### ISSUE TITLE
Pay more attention to element semantics in language checker

### ISSUE THREAD

**Issue by zcorpan** [validator/validator#315]
For this feature https://github.com/validator/validator/issues/284

A doc like

```
<p>the</p><p>the</p><p>the</p><p>the</p><p>the</p> ...
```

is detected as being Albanian, while

```
<p>the</p> <p>the</p> <p>the</p> <p>the</p> <p>the</p> ...
```

is detected as being English.

Semantically there shouldn't be any difference between the two. This is a hypothetical example but with minimizers it's common to strip "insignificant" whitespace, and there have already been a few cases where misdetecting the language causes confusion/trouble for people so I thought it could be good to try to make it a bit better.

I think the code already skips over `script` and `style` and anything not in `body`, but not much else.

Here are some ideas for improvement
## <a id="space-before-after"></a>space before/after:

address, blockquote, center, div, figure, figcaption, footer, form, header, hr,
legend, listing, main, p, pre, xmp,
dialog,
rt,
article, aside, h1, h2, h3, h4, h5, h6, hgroup, nav, section,
dd, dt, li,
caption, th, td,
input, select, option, button, textarea,
fieldset, legend,
embed, iframe, video, canvas, object, audio, img,
details, summary,
marquee,
meter,
progress,
br,
menuitem
## Append attribute value:

img[alt]
## Ignore?

script, style, template, svg, math

Also *[lang] where the language tag (ignoring variant) is different from the root element's lang?
From original issue

> Should support language changes, e.g. do separate analysis for elements with different specified language, and exclude their content from the "parent" analysis.
